Script/AtomicEditor/hostExtensions/languageExtensions/TypscriptLanguageExtension.ts

```diff
@@ -35,71 +52,115 @@ export default class TypescriptLanguageExtension implements Editor.HostExtension
+            const slashedProjectPath = Atomic.addTrailingSlash(ToolCore.toolSystem.project.projectPath);
 
+            const tsconfigFn = Atomic.addTrailingSlash(ToolCore.toolSystem.project.projectPath) + "tsconfig.json";
+            // Let's look for a tsconfig.json file in the project root and add any additional files
+            if (Atomic.fileSystem.fileExists(tsconfigFn)) {
+                // load up the tsconfig file and parse out the files block and compare it to what we have
+                // in resources
+                const file = new Atomic.File(tsconfigFn, Atomic.FILE_READ);
+                try {
+                    const savedTsConfig = JSON.parse(file.readText());
+                    if (savedTsConfig["files"]) {
+                        savedTsConfig["files"].forEach((file: string) => {
+                            let newFile = Atomic.addTrailingSlash(ToolCore.toolSystem.project.projectPath) + file;
+                            let exists = false;
+                            if (Atomic.fileSystem.fileExists(newFile)) {
+                                exists = true;
+                                file = newFile;
+                            } else if (Atomic.fileSystem.exists(file)) {
+                                exists = true;
+                            }
+                            if (exists && projectFiles.indexOf(file) == -1) {
+                                projectFiles.push(file);
+                            }
+                        });
+                    }
 
-            // Look in a 'typings' directory for any typescript definition files
-            const typingsDir = Atomic.addTrailingSlash(ToolCore.toolSystem.project.projectPath) + "typings";
-            Atomic.fileSystem.scanDir(typingsDir, "*.d.ts", Atomic.SCAN_FILES, true).forEach(filename => {
-                projectFiles.push(Atomic.addTrailingSlash(typingsDir) + filename);
-            });
+                    // override the default options if the tsconfig contains them
+                    if (savedTsConfig["compilerOptions"]) {
+                        compilerOptions = savedTsConfig["compilerOptions"];
+                    }
+                } finally {
+                    file.close();
+                }
+            };
 
-            let files = projectFiles.map((f: string) => {
-                if (f.indexOf(ToolCore.toolSystem.project.resourcePath) != -1) {
-                    // if we are in the resources directory, just pass back the path from resources down
-                    return f.replace(Atomic.addTrailingSlash(ToolCore.toolSystem.project.projectPath), "");
-                } else {
-                    // otherwise return the full path
-                    return f;
+            // Then see if we have a copy of Atomic.d.ts in the project directory.  If we don't then we should load it up from the tool environment
+            let found = false;
```

Comment:
The Atomic.d.ts is great, we have some thought to put to versioning and all that, so pleased that this is working out